src/briefcase/integrations/android_sdk.py

```diff
@@ -126,6 +126,64 @@ def verify_android_sdk(command):
     return sdk
 
 
+def _get_choices_and_names(running_devices, all_emulators, avd_name_by_device_id):
+    '''Input:
+
+    - `running_devices`: Dict from ADB device ID to dict with 'model' property.
+    - `all_emulators`: List of emulator device names.
+    - `avd_name_by_device_id`: A dict from ADB device ID to AVD name.
+
+    Output:
+
+    Tuple of `choice_ids`, `full_names`.
+
+    `choice_ids` is a list of all running devices, followed by all non-running
+    AVD names prefixed with `@`.
+
+    `full_names` is dict from string (device ID *or* AVD) to a formatted description
+    of the device.
```

Comment:
Why return both full_names *and* choice_ids? Won't choice_ids == full_names.keys()? (Especially since insertion order is guaranteed in Python 3.6 - and Android won't work on Python 3.5, so we get that one as a gimme.

Reply:
Answering my own question here - it looks like full_names can include more identifiers than is in choice_ids? If that's the case, that's worth clarifying in the explanation here.